witness.schema.json

```diff
@@ -0,0 +1,393 @@
+        },
+        "type": {
+          "description": "Interpretation of `string`. The following values are supported:\n\n* `assertion`: Assertion, e.g. `assert(<string>)` in C, inserted at the specified location. The insertion is hypothetical and doesn't affect other locations.",
+          "enum": [
+            "assertion"
+          ]
+        },
+        "format": {
+          "description": "Format of `string`. The following values are supported:\n\n* `C`: Expression in C language.",
+          "enum": [
+            "C"
+          ]
+        }
+      }
+    },
+    "target": {
+      "description": "Target entry of certification.",
+      "type": "object",
+      "required": [
+        "uuid",
+        "type",
+        "file_hash"
+      ],
+      "properties": {
+        "uuid": {
+          "$ref": "#/$defs/entry-uuid",
+          "description": "Unique identifier of the target entry."
+        },
+        "type": {
+          "type": "string",
+          "description": "`entry_type` of the target entry."
+        },
+        "file_hash": {
+          "$ref": "#/$defs/hash",
+          "description": "SHA-256 hash of the target file."
+        }
+      }
+    },
+    "certification": {
+      "description": "Certification result.",
+      "type": "object",
+      "required": [
+        "string",
+        "type",
+        "format"
+      ],
+      "properties": {
+        "string": {
+          "type": "string",
+          "description": "The certification result itself."
+        },
+        "type": {
+          "description": "Interpretation of `string`. The following values are supported:\n\n* `verdict`: Verdict about the targeted entry.",
+          "enum": [
+            "verdict"
+          ]
+        },
+        "format": {
+          "description": "Format of `string`. The following values are supported:\n\n* `confirmed | rejected`: Either \"confirmed\" or \"rejected\".",
+          "enum": [
+            "confirmed | rejected"
+          ]
+        }
+      }
+    }
+  },
+  "type": "array",
+  "items": {
+    "oneOf": [
+      {
+        "title": "Loop invariant",
+        "description": "Loop invariants are important building blocks in software verification. Many verification approaches use loop invariants as lemmata to construct proofs of correctness.",
+        "type": "object",
+        "required": [
+          "entry_type",
+          "metadata",
+          "location",
+          "loop_invariant"
+        ],
+        "properties": {
+          "entry_type": {
+            "const": "loop_invariant"
+          },
+          "metadata": {
+            "$ref": "#/$defs/metadata"
+          },
+          "location": {
+            "$ref": "#/$defs/location",
+            "description": "Location in the source code where the loop invariant holds."
+          },
+          "loop_invariant": {
+            "$ref": "#/$defs/invariant",
+            "description": "The loop invariant itself."
```

Comment:
Not completely accurate, since `loop_invariant` also contains the `type` and `format`. The actual loop invariant is stored in `loop_invariant.string`, so here it is perhaps better to write
```
Description of the loop invariant.
```